hazelcast-sql/src/main/java/com/hazelcast/sql/impl/calcite/opt/physical/visitor/PlanCreateVisitor.java

```diff
@@ -277,9 +277,15 @@ private static SqlRowMetadata createRowMetadata(List<String> columnNames, List<Q
 
     @Override
     public void onMapScan(MapScanPhysicalRel rel) {
-        HazelcastTable hazelcastTable = rel.getTableUnwrapped();
         AbstractMapTable table = rel.getMap();
 
+        if (table.nativeMemoryEnabled()) {
+            throw QueryException.error("IMap scan with InMemoryFormat.NATIVE is not supported: " + table.getName()
```

Comment:
The error message is not very intuitive, because from the user perspective there is no "scan", but instead, there is an access to a table. I would change the message to something like this:
```
Cannot query IMap <name> with InMemoryFormat.NATIVE because it does not have indexes (please add at least one index to query this IMap)
```